packages/tools/client-logger/app-insights-logger/src/FluidAppInsightsLogger.ts

```diff
@@ -0,0 +1,53 @@
+import { ApplicationInsights, Snippet } from "@microsoft/applicationinsights-web";
+
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { ITelemetryBufferedLogger } from "@fluidframework/test-driver-definitions";
+
+/**
+ * Config object for FluidAppInsightsLogger.
+ */
+export interface FluidAppInsightsLoggerConfig {
+	/**
+	 * Accepts either an existing preconfigured ApplicationInsights instance to use
+	 * for logging or a config object used to instantiate an instance of ApplicationInsights. If providing a
+	 * preconfigured instance of ApplicationInsights, .loadAppInsights() must be called on the instance first.
+	 */
+	appInsights: ApplicationInsights | Snippet;
+}
+
+export class FluidAppInsightsLogger implements ITelemetryBufferedLogger {
+	protected readonly baseLoggingClient: ApplicationInsights;
+
+	/**
+	 * Creates an instance of FluidAppInsightsLogger using either a provided instance of ApplicationInsights
+	 * logging client or by creating a new instance with a provided ApplicationInsights configuration object.
+	 */
+	public constructor(config: FluidAppInsightsLoggerConfig) {
+		if (config.appInsights instanceof ApplicationInsights) {
+			this.baseLoggingClient = config.appInsights;
+		} else {
+			if (config.appInsights === undefined) {
+				throw Error(
+					"Cannot initialize default AppInsights Client without a configuration object.",
+				);
+			}
+			this.baseLoggingClient = new ApplicationInsights(config.appInsights);
+			this.baseLoggingClient.loadAppInsights();
+		}
+	}
+
+	getBaseLoggingClient() {
+		return this.baseLoggingClient;
+	}
+
+	send(event: ITelemetryBaseEvent): void {
+		this.baseLoggingClient.trackEvent({
+			name: event.eventName,
+			properties: event,
+		});
+	}
+
+	async flush(): Promise<void> {
```

Comment:
this is never called. nothing in fluid will call it, so why even have it.

Reply:
The example should call it, that's true; and documentation should make it clear why consumers should make a best effort to call it.

@anthony-murphy we already know for a fact that not flushing the underlying ApplicationInsights client can result in some messages not making it to AppInsights. I don't know that we (or consumers) can do much in the case of a user just closing the application tab, but I think we should keep it and document it, so consumers know that it's supposed to be called when possible, and what can they expect if for whatever reason it is not.

Reply:
i get it. just calling out that the pattern here isn't leveraged. Another options is to not have it here, but call out the owner of the ApplicationInsights logger that is passed in, so the application, should call flush. I'm not a big fan of defining an abstraction just to wrap something.